File: algorithms/278. First Bad Version/solution.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution(object):
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        low, high = 1, n
        while low < high:
            mid = low + (high - low) / 2
            if isBadVersion(mid):
                high = mid
            else:
                low = mid + 1
        return low

File: algorithms/278. First Bad Version/solution.cc
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int low = 1, high = n, mid;
        while ( low < high ) {
            mid = low + (high - low) / 2;
            if ( isBadVersion(mid) ) {
                high = mid;
            } else {
                low = mid + 1;
            }
        }
        return low;
    }
};
